Dockerfile

```diff
@@ -1,42 +1,45 @@
-FROM registry.access.redhat.com/ubi8/ubi
-LABEL maintainer="James Harmison <jharmison@redhat.com>"
+# Downloader will retrieve installation bundles
+FROM registry.access.redhat.com/ubi8/ubi as downloader
+ARG SONARQUBE_VERSION=8.7.0.41497
+ARG SONAR_JAVA_PLUGIN_VERSION=6.13.0.25138
+
+ENV SONARQUBE_VERSION=$SONARQUBE_VERSION \
+    SONAR_JAVA_PLUGIN_VERSION=$SONAR_JAVA_PLUGIN_VERSION
 
-ARG SONAR_VERSION=7.9.4
-ARG BUILD_DATE=2020-08-06
+WORKDIR /download
+COPY scripts/sonar-download.sh ./
+RUN dnf -y install unzip curl gpg \
+ && ./sonar-download.sh
 
-ENV SONAR_VERSION=$SONAR_VERSION \
-    SONARQUBE_HOME=/opt/sonarqube
+# Final image will have Sonarqube installed
+FROM registry.access.redhat.com/ubi8/ubi
+ARG SONARQUBE_VERSION=8.7.0.41497
+ARG SONAR_JAVA_PLUGIN_VERSION=6.13.0.25138
+
+ENV SONARQUBE_VERSION=$SONARQUBE_VERSION \
+    SONAR_JAVA_PLUGIN_VERSION=$SONAR_JAVA_PLUGIN_VERSION
 
 LABEL name="SonarQube" \
+      vendor="Sonatype" \
       io.k8s.display-name="SonarQube" \
-      io.k8s.description="Provide a SonarQube image to run on Red Hat OpenShift" \
-      io.openshift.expose-services="9000" \
+      io.openshift.expose-services="9000:http" \
       io.openshift.tags="sonarqube" \
-      build-date=$BUILD_DATE \
-      version=$SONAR_VERSION \
-      release="1"
-
-
-RUN yum -y install unzip java-11-openjdk nss_wrapper nodejs \
-    && yum clean all \
-    && rm -rf /var/cache/yum \
-    && cd /tmp \
-    && curl -o sonarqube.zip -fSL https://binaries.sonarsource.com/Distribution/sonarqube/sonarqube-$SONAR_VERSION.zip \
-    && cd /opt \
-    && unzip /tmp/sonarqube.zip \
-    && mv sonarqube-$SONAR_VERSION sonarqube \
-    && rm /tmp/sonarqube.zip* \
-    && rm /opt/sonarqube/extensions/plugins/sonar-java-plugin-*.jar \
-    && curl -o /opt/sonarqube/extensions/plugins/sonar-java-plugin-6.3.2.22818.jar https://binaries.sonarsource.com/Distribution/sonar-java-plugin/sonar-java-plugin-6.3.2.22818.jar
-
-ADD root /
-
-RUN useradd -r sonar \
-    && chmod 775 $SONARQUBE_HOME/bin/run_sonarqube.sh \
-    && /usr/bin/fix-permissions /opt/sonarqube
-
-USER sonar
-WORKDIR $SONARQUBE_HOME
-VOLUME $SONARQUBE_HOME/data
+      org.sonarqube.version=$SONARQUBE_VERSION \
+      org.sonarqube.plugins.sonar-java.version=$SONAR_JAVA_PLUGIN_VERSION \
+      release="2" \
+      maintainer="James Harmison <jharmison@redhat.com>"
+
+COPY --from=downloader /download/opt /opt
+RUN dnf -y install java-11-openjdk nodejs \
```

Comment:
Would it be easier for us to derive this layer from the `ubi8/openjdk-11` base image instead of installing OpenJDK here? Although that brings in the s2i scripts too, I think.

Reply:
Let's check how big of a difference it makes:

```bash
podman run -it --name java registry.access.redhat.com/ubi8/ubi bash -c 'dnf -y install java-11-openjdk nodejs && dnf clean all && rm -rf /var/cache/yum /var/cache/dnf'
podman commit java java-ubi-only
```

I expected the difference to be more dramatic than it is, but this is roughly why I'd rather inherit the base UBI (which is common enough a base to be well understood):

```shell
$ podman images
REPOSITORY                                   TAG            IMAGE ID      CREATED         SIZE
localhost/java-ubi-only                      latest         c3f9627656c9  13 seconds ago  595 MB
registry.access.redhat.com/ubi8/openjdk-11   latest         9be97c3f2a59  10 days ago     630 MB
registry.access.redhat.com/ubi8/ubi          latest         4199acc83c6a  5 weeks ago     213 MB
```

Base UBI is 213 MB, the openjdk-11 image (which has S2I, but also a lot of other "suggested" things) is 630, and our UBI + just the things we need for SonarQube is 595MB.

In the case of some of our other tools especially the difference is dramatic. Java just is this way, I suppose. :) I still don't think we need to S2I tooling to end up here, though.

Thanks for the :eyes: !